soc/renesas/ra/ra4m1/Kconfig

```diff
@@ -9,3 +9,26 @@ config SOC_SERIES_RA4M1
 	select DYNAMIC_INTERRUPTS
 	select TIMER_READS_ITS_FREQUENCY_AT_RUNTIME
 	select XIP
+
+config SOC_OPTION_SETTING_MEMORY
+	bool "Option Setting Memory"
+	default y
+
+config SOC_OPTION_SETTING_LVDAS_DISABLE
+	bool "Disable voltage monitor 0 after reset"
+	default y
+	depends on SOC_OPTION_SETTING_MEMORY
+
+config SOC_OPTION_SETTING_HOCO_DISABLE
+	bool "Disable HOCO oscillator after reset"
+	depends on SOC_OPTION_SETTING_MEMORY
+
+config SOC_OPTION_SETTING_HOCO_FREQ
+	int "HOCO oscillator frequency"
+	default $(dt_node_int_prop_int,/clocks/hoco,clock-frequency)
+	depends on SOC_OPTION_SETTING_MEMORY
```

Comment:
We can use the devicetree definition to determine hoco frequency by the following code.

```
DT_PROP(DT_PATH(clocks, hoco), clock_frequency)
```

Reply:
hi @soburi, i'm new to zephyr so i could be wrong about this but is seems like I can't use:

```DT_PROP(DT_PATH(clocks, hoco), clock_frequency)```

inside a Kconfig file to access a devicetree property, however i can use this within a 'C' file to access a devicetree property.

when i check other Kconfig files that refer to the devicetree they all seem to use:

```dt_node_xxx_yyy```

see soc/nxp/mcx/mcxnx4x/Kconfig for an example

let me know what you think, I probably missed something obvious :)

Reply:
I think this part ...

https://github.com/zephyrproject-rtos/zephyr/pull/71017/files#diff-7cd037a8cbd8c3e02942d0cbda9cf90aa08fbab8ab55dcf57b46e12f660662beR8-R18

can rewrite as follows.
We can refer the hoco clock frequency directly in code, so we no need to define hoco freq in Kconfig.

```
#define HOCO_FREQ DT_PROP(DT_PATH(clocks, hoco), clock_frequency)

#if HOCO_FREQ == 24000000
#define OFS1_HOCO_FREQ		0
#elif HOCO_FREQ == 32000000
#define OFS1_HOCO_FREQ		2
#elif HOCO_FREQ == 48000000
#define OFS1_HOCO_FREQ		4
#elif HOCO_FREQ == 64000000
#define OFS1_HOCO_FREQ		5
#else
#error "Unsupported HOCO frequency"
#endif
```